website_sale_affiliate/controllers/main.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl)
+import logging
+
+from odoo.http import request, route
+from odoo.addons.website_sale.controllers.main import WebsiteSale as Base
+
+_logger = logging.getLogger(__name__)
+
+
+class WebsiteSale(Base):
+    def _store_affiliate_info(self, **kwargs):
```

Comment:
Thanks for the lively discussions! I appreciate the explanations/arguments for each. My nested try/except statements were pretty ugly, and either of your suggested changes would be an improvement. I'm going with the flatter try/except suggestion here. I think it's easier to follow, and though the key is useless without an ID, it's also harmless.